### PR TITLE
checker(dm): support wildcard in privilege checking

### LINKED ISSUE #7645: privilege checker doesn't support wildcard database permissions
### What did you do?

Task:
```yaml
name: demo-mysql8-tidb
task-mode: "all"
mysql-instances:
  - source-id: "mysql8"
    block-allow-list: "mysql8-tidb-test"

target-database:
  host: "127.0.0.1"
  port: 4000
  user: "root"
  password: ""

block-allow-list:
  mysql8-tidb-test:
    do-dbs: ["demo_foobar"] 
```

Source DB privileges:
```
mysql-8.0.31> SHOW GRANTS FOR 'dmuser'@'%';
+--------------------------------------------------------------------+
| Grants for dmuser@%                                                |
+--------------------------------------------------------------------+
| GRANT REPLICATION SLAVE, REPLICATION CLIENT ON *.* TO `dmuser`@`%` |
| GRANT SELECT ON `demo_%`.* TO `dmuser`@`%`                         |
+--------------------------------------------------------------------+
2 rows in set (0.00 sec)
```

check-task output:
```json
{
    "result": false,
    "msg": "[code=26005:class=dm-master:scope=internal:level=medium], Message: fail to check synchronization configuration with type: check was failed, please see detail
    	detail: {
		"results": [
			{
				"id": 3,
				"name": "source db dump privilege checker",
				"desc": "check dump privileges of source DB",
				"state": "fail",
				"errors": [
					{
						"severity": "fail",
						"short_error": "lack of Select privilege: {`demo_foobar`.`t1`}; "
					}
				],
				"instruction": "You need grant related privileges.",
				"extra": "address of db instance - 127.0.0.1:8031"
			},
			{
				"id": 2,
				"name": "mysql_version",
				"desc": "check whether mysql version is satisfied",
				"state": "warn",
				"errors": [
					{
						"severity": "warn",
						"short_error": "version suggested less than 8.0.0 but got 8.0.31"
					}
				],
				"extra": "address of db instance - 127.0.0.1:8031"
			}
		],
		"summary": {
			"passed": false,
			"total": 12,
			"successful": 10,
			"failed": 1,
			"warning": 1
		}
	}"
}

```

### What did you expect to see?

Permission check to succeed.

### What did you see instead?

```
lack of Select privilege: {`demo_foobar`.`t1`};
```

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
v6.5.0 master at 9d60d064ef5e681867a60c95cdb2b99c10c0c67c
```
Upstream MySQL/MariaDB server version:
```console
8.0.31
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
v6.1.2
```
How did you deploy DM: tiup or manually?
```console
local binary
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close #7645

### What is changed and how it works?

use the same function of TiDB privilege checking

#### Tests 

 - Unit test

### Release note 

```release-note
Fix a bug that DM may raise error at prechecking when downstream database name in SHOW GRANTS contains wildcard
```

### CODE REVIEW

**Review by buchuitoudegou — APPROVED** [review]
lgtm

File: dm/pkg/checker/privilege.go [hunk 1]
```diff
@@ -369,34 +372,36 @@ func VerifyPrivileges(
 						if privs.needGlobal {
 							continue
 						}
-						dbPrivs, ok := privs.dbs[dbName]
-						if !ok || dbPrivs.wholeDB {
-							continue
-						}
-						if _, ok := dbPrivs.tables[tableName]; !ok {
-							continue
+						for dbName, dbPrivs := range privs.dbs {
```
Comment (D3Hunter) [inline dm/pkg/checker/privilege.go:375]: > The _ and % wildcards are permitted when specifying database names in [GRANT](https://dev.mysql.com/doc/refman/8.0/en/grant.html) statements that grant privileges at the database level (GRANT ... ON db_name.*).

wildcard cannot be used on table level grant, not need to change here

Comment (lance6716) [reply dm/pkg/checker/privilege.go]: https://github.com/pingcap/tiflow/pull/7739/commits/8dc362cba5951ae397db60748b733c04f042fa5a

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**